tests/js/spec/views/organizationIncidents/details/activity.spec.jsx

```diff
@@ -24,34 +33,191 @@ describe('IncidentDetails -> Activity', function() {
       routerContext
     );
 
-  it.todo('fetches and renders activities');
+  it('fetches and renders activities', async function() {
+    const wrapper = createWrapper();
+
+    expect(activitiesList).toHaveBeenCalled();
+
+    // loading
+    expect(wrapper.find('Placeholder').length).toBeGreaterThan(0);
+
+    await tick();
+    wrapper.update();
+
+    expect(wrapper.find('Placeholder')).toHaveLength(0);
+    expect(
+      wrapper
+        .find('NoteBody')
+        .text()
+        .trim()
+    ).toEqual('incident activity comment');
+  });
+
   it('creates a new note', async function() {
-    const createNote = MockApiClient.addMockResponse({
+    const createComment = MockApiClient.addMockResponse({
       url: `/organizations/${organization.slug}/incidents/${
         incident.identifier
       }/comments/`,
       method: 'POST',
       body: TestStubs.IncidentActivity({
+        id: '234',
         comment: 'new incident comment',
       }),
     });
 
     const wrapper = createWrapper();
 
     changeReactMentionsInput(wrapper, 'new incident comment');
-    // wrapper.find('NoteInput Button[type="submit"]').simulate('click');
     wrapper.find('textarea').simulate('keyDown', {key: 'Enter', ctrlKey: true});
 
     await tick();
-    expect(createNote).toHaveBeenCalledWith(
+    expect(createComment).toHaveBeenCalledWith(
       `/organizations/${organization.slug}/incidents/${incident.identifier}/comments/`,
       expect.objectContaining({data: {comment: 'new incident comment'}})
     );
   });
 
   it.todo('fails to create a new note');
-  it.todo('updates an existing note');
-  it.todo('fails to update an existing note');
-  it.todo('deletes a note');
-  it.todo('fails to delete a note');
+
+  it('updates an existing note', async function() {
+    const newComment = 'edited comment';
+    const updateComment = MockApiClient.addMockResponse({
+      url: `/organizations/${organization.slug}/incidents/${
+        incident.identifier
+      }/comments/${activity.id}/`,
+      method: 'PUT',
+      body: {
+        ...activity,
+        comment: newComment,
+      },
+    });
+
+    const wrapper = createWrapper();
+
+    await tick();
+    // unfortunately edit/delete items are hidden until hover (using emotion)
+    // so we can't simulate this in jest?
```

Comment:
Do we need to worry about how those hover interactions are reached on a touch device?

Reply:
We probably should, this is also the same behavior as in our Issue comments